jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -108,13 +113,6 @@ def _on_viewer_data_changed(self, msg=None):
         self._viewer_id = self.app._viewer_item_by_reference(
             'spectrum-viewer').get('id')
 
-        # Subsets are global and are not linked to specific viewer instances,
-        # so it's not required that we match any specific ids for that case.
-        # However, if the msg is not none, check to make sure that it's the
-        # viewer we care about.
-        if msg is not None and msg.viewer_id != self._viewer_id:
```

Comment:
Do we not need this?

Reply:
No, and in fact it was breaking it because now it's listening for messages that don't have a `viewer_id` attached. This was a minuscule performance improvement to not update the data dropdown on irrelevant `AddData` messages.